Content.Shared/Storage/SharedItemCounterSystem.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using JetBrains.Annotations;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+
+namespace Content.Shared.Storage
+{
+    [UsedImplicitly]
+    public abstract class SharedItemCounterSystem : EntitySystem
+    {
+        /// <inheritdoc />
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeNetworkEvent<AfterStorageFillEvent>(OnStorageFill);
```

Comment:
uhhhhh this subscribes to every single storage fill event raised anywhere, does this even need to be a network event (both storagefill and itemcounter run on the server, presumably?)

really this should just be
```c#
SubscribeLocalEvent<SharedItemCounterComponent, AfterStorageFillEvent>(OnStorageFill);
```

Reply:
Subscribing Locally doesn't work and leads to sync issues. 

I tried your approach with two clients. I spawned crayon in one client and other didn't see it. Same with changing and removing crayons. While box content is the same the visualization doesn't work.